### ISSUE TITLE
PS TV No way to disable bilinear filtering on PS1 games?

### ISSUE THREAD

**Issue by TheStremon** [TheOfficialFloW/Adrenaline#424]
I'm sorry if a solution or answer as already been given, I searched everywhere and I couldn't find anything.

On the PS TV for PS1 games, the official filter is activated by default, and there is no way to disable it, making the games look very bad and blurred.
I tried all the options in adrenaline, nothing removes the blur.

What's even more strange is that when starting a PS1 game, the PS1 logo displayed at the beginning doesn't seem to be filtered, the filter is only activated afterward.

Is there any way to manually force this filter off?
I am using Adrenaline 6.9 on a 3.60 Enzo PS TV

**Comment by rsn8887** [comment]
Use filter: sharp bilinear without Scanlines
Then Turn smooth graphics on or off however you like it.

I like it setting it on that’s how it was designed.

**Comment by TheStremon** [comment]
That's the issue, using sharp bilinear without scanline and disabling filter makes almost no visible difference on the PS1 games of the PS TV, it still has a very blurry image, way more blurry than on the Vita. 

Other things like psp games are not blurry like that, and I can see a real change on them when chaging filter options. Everything works nicely on the vita. It just doesn't work on the PS TV.

The problem is that on the PS TV the original filter seems to be active by default even when using other adrenaline filters, and disabling it in the adrenaline options makes a very small difference since it only disable adrenaline filter, not the internal one. You can see the internal filter is always active no matter what, and activating the adrenaline filter in the option only add another similar filter in the top if it.

And I don't think PS1 games are supposed to look like that, the same PSN games never look that blurry on PS3 or PSP Go on the same screen, no matter which option I use.
Also it's a matter of preference, of taste, it would be great to have the choice to see those fat pixels or not, like on PS Vita or PSP 😉

**Comment by rsn8887** [comment]
My recommended settings
VitaTV:
````
(Vita TV has no official settings for display options)

Adrenaline menu settings
=====
Graphics filtering: Sharp bilinear without scanlines
Smooth graphics: On
Screen Scale X (PSP): 2.000 (or 1.995 if you like it blurry)
Screen Scale Y (PSP): 2.000 (or 1.995 if you like it blurry)
Screen Scale X (PS1): 1.185
Screen Scale Y (PS1): 1.000
````

I think what you are seeing is simply your TV upscaling the VitaTV 720p output to its own 1080p resolution. There’s no way around that because VitaTV cannot output 1080p, only 720p.

It might be good to double check that your VitaTV is at least set to 720p output, not 480p or 1080i, which look much worse than 720p. There’s a setting for that on VitaTV some kind of hdmi setting or so.

**Comment by TheStremon** [comment]
Yes, the settings you are giving me are the first ones I tried.
For the 720p, yes my PSTV is setup to 720p.
Luckily, my TV is upscaling 720p content very nicely, it's very sharp. The Snes mini for instance, that is also in 720p, looks great and sharp with every filter disabled. And even on the PSTV, the PSP games and menu look ok too.
Very pixelated, but that's the way I like it.

Thank you for the explanation you give me, but I already know all that, I've been in the emulation world and in PSP modding pretty much since the very beginning, making my own pops games and all. Which makes me feel so old when I think about it xD

I do know that no matter what option I choose, the PSTV won't be as sharp as the Vita or PSP screen, for the reasons you tell me, and also because the content of the PSTV is basically the Vita 960x544 resolution, upscaled to 720p.

However, I am not searching for the perfect sharp solution, I just want to be able to disable the internal bilinear filter, that makes everything way too blurry for my taste. That's all. I know many people, like you for instance, prefer it with the filter active, or don't mind about it, which I totally respect, but many retrogamers like me prefer the good old sharp pixels. Just watch the video of My Life In Gaming about the PS1 solutions if you don't believe me ;)

I am 100% sure that the blur I have comes mostly from this official bilinear filtering, since when I activate the option in the official settings of the Vita, I got the exact same blur, except it's worse on the PSTV for the reasons we mentioned.
But then when I disable the filter on the Vita, everything becomes so sharp and nice, as expected, even with the Adrenaline smooth graphics it's still quite sharp. I do use the original scaling option on the vita, not the adrenaline one.

For the 480i games you mention, it's weird, for me they looks great. 
But that's probably because you use the Adrenaline screen scale instead of the native scale. On the Vita when I use the "original" or "normal" scale modes in the official setting, without any filter, the games look great and sharp. Even the manual scale still upscales everything without so much distortion or blur. But when I use Adrenaline screen scale, without filter everything gets indeed so messed up. Those scaling are obviously not working the same way, the official one is a pre scaling that changes only the PS1 output, trying to scale as even as possible, whereas the adrenaline one is a post scaling, pretty much scaling the whole display image unevenly.
The PSX games on Vita are displayed in native 480 pixel height (x1 for 480i games and x2 for 240p games), allowing to fit nicely in the Vita 544 pixel height with black borders, so of course 480i games will look more messed up by the uneven scaling, because of their double resolution.

So the question I asked in the title of my post still stands, is it possible in any way to force disable the official bilinear filtering on the PSTV? After extensive search, I saw many other people asking the same thing over the years, but it was never truly addressed.

**Comment by MapleStoryPSN** [comment]
Would also like to chime in on how bad some graphics look on PSTV. The sprites and textures in Tactics Ogre: Let Us Cling Together (PSP) look horribly muddy and out of place compared to how sharp pixels look on the PSP or even emulation (PPSSPP). This applies to playing it on a handheld Vita as well.

**Comment by rsn8887** [comment]
The official bilinear filter seems to perform a shift of the pixels. It might be best to compare by comparing screenshots, but I do not have my Vita atm. Last time I investigated this, I realized that the shift  messed up PSX 480i games when the official bilinear filter was off. It was easy to tell by loading gran turismo ps1 and looking at the font characters in the 480i menu screen.

I only tested this with scale 1.135 on handheld Vita. You are probably right that there’s no such issue with scale 1.000.

**Comment by TheStremon** [comment]
I actually want to do a series of screenshots on both my Vita and PSTV showing the results of every mode and resolution, just to compare what does what. I will do it soon and upload it here, it could be useful 😉

Meanwhile I would really like to know how the adrenaline scaler algorithm is choosing the pixel position and all, compared to the official manual scaling, to see what can be done to make it work better on the PSTV.
I could be wrong, but for me it looks like that without filter, the adrenaline scaling is working like a "closest neighbor" algorithm, while the official scaler seems to limit the scaling and moving to larger steps than adrenaline, to avoid messing up the look too much by having pixels all over the place.
Don't know if my explanation makes sense for you guys or not 😂

Also since it was possible to make a plug-in to emulate the behavior of the PSTV on the Vita (for DS3/DS4 controllers), I wonder if it's possible to do the other way around to get more screen options. Not sure it's doable since the PSTV is not handling the video modes the same way.

**Comment by rsn8887** [comment]
I am not sure about your last question, but that should be easy to test. Be sure to post any test screenshots you make here I would love to take a look, too. We might be able to get to the bottom of some of this (official bilinear vs Adrenaline bilinear etc)

**Comment by TheStremon** [comment]
Here are the screenshots on the Vita on psx in 240p:

`Official settings
=====
Bilinear Filtering: Off
Screen Mode: Original

Adrenaline menu settings
=====
Graphics filtering: Bilinear
Smooth graphics: Off
Screen Scale X (PS1): 1.000
Screen Scale Y (PS1): 1.000`
![2019-05-14-004916](https://user-images.githubusercontent.com/16698212/57639526-0f8a5880-75eb-11e9-8033-642158901176.jpg)

`Official settings
=====
Bilinear Filtering: Off
Screen Mode: Normal (4/3 scale)`
![2019-05-14-005018](https://user-images.githubusercontent.com/16698212/57639947-15346e00-75ec-11e9-94fd-f9f8ab6d3940.jpg)

`Official settings
=====
Bilinear Filtering: On
Screen Mode: Original`
![2019-05-14-005051](https://user-images.githubusercontent.com/16698212/57640099-7e1be600-75ec-11e9-8f7d-b31a0cfab008.jpg)

`Official settings
=====
Bilinear Filtering: On
Screen Mode: Normal (4/3 scale)`
![2019-05-14-005107](https://user-images.githubusercontent.com/16698212/57640167-ae638480-75ec-11e9-891d-70a514e5b625.jpg)

`Official settings
=====
Bilinear Filtering: Off
Screen Mode: Original

Adrenaline menu settings
=====
Graphics filtering: Sharp bilinear without scanlines
Smooth graphics: On
Screen Scale X (PS1): 1.000
Screen Scale Y (PS1): 1.000`
![2019-05-14-005329](https://user-images.githubusercontent.com/16698212/57640265-fda9b500-75ec-11e9-9954-f4f039d2bf7f.jpg)

`Adrenaline menu settings
=====
Graphics filtering: Sharp bilinear without scanlines
Smooth graphics: Off
Screen Scale X (PS1): 1.135
Screen Scale Y (PS1): 1.135`
![2019-05-14-005428](https://user-images.githubusercontent.com/16698212/57640400-6db83b00-75ed-11e9-949f-4e8da0e20f04.jpg)

`Adrenaline menu settings
=====
Graphics filtering: Sharp bilinear without scanlines
Smooth graphics: On
Screen Scale X (PS1): 1.135
Screen Scale Y (PS1): 1.135`
![2019-05-14-005500](https://user-images.githubusercontent.com/16698212/57640445-8aed0980-75ed-11e9-8e14-128e035f25f9.jpg)

I first tried the official video modes only, with original and normal modes, original being pixel perfect and normal the scaled 4/3. Then the same modes with the bilinear filtering
After that, I tried the adrenaline option we talked about.

To sum up, the official scale is giving a better result than the adrenaline one, however the official bilinear filter blurs everything out, where the adrenaline smooth option is a lot better, smoothing only when needed, and giving better results for uneven stretch.

So as we mentioned, the PSTV has the bilinear filter always on, even when using adrenaline filters. So since those filters are post filters adding up to the official one, no matter which option we choose it will always be a muddy blurry mess 😝

I will try doing comparative screenshots on the PSTV, and also for 480i games.
But somehow the vita has trouble taking screenshots of the 480i, half of the ones I took we're corrupted...

**Comment by TheStremon** [comment]
Here are the pics from the PSTV:

Adrenaline menu settings
=====
Graphics filtering: Bilinear
Smooth graphics: Off
Screen Scale X (PS1): 1.000
Screen Scale Y (PS1): 1.000
![2019-05-18-171505](https://user-images.githubusercontent.com/16698212/57966937-921c6a80-7993-11e9-8e8d-bd348139c77b.jpg)

Adrenaline menu settings
=====
Graphics filtering: Sharp bilinear without scanlines
Smooth graphics: On
Screen Scale X (PS1): 1.000
Screen Scale Y (PS1): 1.000
![2019-05-18-171516](https://user-images.githubusercontent.com/16698212/57966946-a7919480-7993-11e9-9ed2-f7d151dee5d4.jpg)

Adrenaline menu settings
=====
Graphics filtering: Sharp bilinear without scanlines
Smooth graphics: Off
Screen Scale X (PS1): 1.185
Screen Scale Y (PS1): 1.000
![2019-05-18-172954](https://user-images.githubusercontent.com/16698212/57966956-d445ac00-7993-11e9-8fba-142821e69fbc.jpg)

Adrenaline menu settings
=====
Graphics filtering: Sharp bilinear without scanlines
Smooth graphics: On
Screen Scale X (PS1): 1.185
Screen Scale Y (PS1): 1.000
![2019-05-18-173004](https://user-images.githubusercontent.com/16698212/57966974-f2aba780-7993-11e9-98fa-cc4cf49fe81c.jpg)

So as you can see, it's clear the bilinear filter is On by default no matter what option is used.
Note that the captured images are in the Vita native resolution, not in 720p, as expected since the PSTV doesn't use native 720p, at least for the PSP/PSX emulation.
Also again, the smooth graphic option is only useful when using Adrenaline stretching,

**Comment by rsn8887** [comment]
I think the 240p PS1 games are always pre-scaled internally to 480p before Adrenaline sees them. Adrenaline always sees a 640x480 PS1 framebuffer, never 320x240.

The official bilinear filter works by creating new pixels. Each new pixel corresponds to the average of four pixels in the original image. Since the new pixels correspond to the average at the centers of squares of four, the image is shifted by half a pixel top-and-left. This allows a symmetric filtering, where everything is centered.

Official Bilinear OFF
<img width="436" alt="OfficialBilinear_OFF" src="https://user-images.githubusercontent.com/13071547/57983683-7c9e5380-7a1a-11e9-8c34-354a11bf0062.png">
Official Bilinear ON
<img width="436" alt="OfficialBilinear_ON" src="https://user-images.githubusercontent.com/13071547/57983677-74461880-7a1a-11e9-8ffc-8f88e19af27d.png">

When 640x480 PS1 games are played, one would think the bilinear filtering would do nothing. But on my Handheld Vita, it does something. The image pixels are very distorted unless I turn official bilinear filtering on. I tested this with the Gran Turismo Menu and Tobal No 1.

**Comment by TheStremon** [comment]
Yes, the thing about the game being prescaled is exactly what I said few posts ago. I do know how bilinear filter works, thank you 😄
They are distorted even with adrenaline scale to 1.0?
Because the few games I have in 480i (Chrono cross main menu for instance) don't seem distorted, with or without bilinear filter. But with with adrenaline scale it messes up everything because unlike the official scaling function, it doesn't snap the pixels to the nearest pixel in a clean way, it scales unevenly, so no matter what you do, without filter you won't have good result.
It's the same on 240p games, but since they have half the pixels, it's less visible.
I also suspect the interlacing of 480i sources of making the games look a bit worse somehow.

I will do more test on 480i games to see what's going on.

Meanwhile, let's please get back to the main issue, finding a way to disable the official bilinear filter on the PSTV. Because no matter our preferences, having a way to disable and enable it would be great.

**Comment by TheStremon** [comment]
So no way of fixing it I guess? Should I give up on it already? 😅
